estimagic/tests/visualization/test_comparison_plot.py

```diff
@@ -0,0 +1,404 @@
+        "final_value"
+    ] + 0.5 * np.array(diff1)
+    res_dict["mod4"]["result_df"]["conf_int_lower"] = res_dict["mod4"]["result_df"][
+        "final_value"
+    ] - 0.5 * np.array(diff2)
+
+    return res_dict
+
+
+@pytest.fixture()
+def df():
+    with open("estimagic/tests/visualization/minimal_expected_df.json", "r") as f:
+        df = pd.DataFrame(json.load(f))
+        df["conf_int_upper"] = pd.np.nan
+        df["conf_int_lower"] = pd.np.nan
+    return df
+
+
+# _prep_result_df
+# ===========================================================================
+
+
+def test_prep_result_df(minimal_res_dict):
+    model = "mod1"
+    model_dict = minimal_res_dict[model]
+    res = _prep_result_df(model_dict, model)
+    expected = pd.DataFrame.from_dict(
+        {
+            "group": {0: "a", 1: "a", 2: "b", 3: "b", 4: "b", 5: "b", 6: "c"},
+            "name": {0: "u", 1: "i", 2: "a", 3: "e", 4: "n", 5: "r", 6: "t"},
+            "final_value": {
+                0: 1.58,
+                1: 2.01,
+                2: 2.73,
+                3: 1.62,
+                4: 2.18,
+                5: 1.75,
+                6: 2.25,
+            },
+            "index": {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6},
+            "full_name": {
+                0: "a_u",
+                1: "a_i",
+                2: "b_a",
+                3: "b_e",
+                4: "b_n",
+                5: "b_r",
+                6: "c_t",
+            },
+            "model_class": {
+                0: "no class",
+                1: "no class",
+                2: "no class",
+                3: "no class",
+                4: "no class",
+                5: "no class",
+                6: "no class",
+            },
+            "model": {
+                0: "mod1",
+                1: "mod1",
+                2: "mod1",
+                3: "mod1",
+                4: "mod1",
+                5: "mod1",
+                6: "mod1",
+            },
+        }
+    )
+    assert_frame_equal(res, expected, check_like=True)
+
+
+# _df_with_all_results
+# ===========================================================================
+
+
+def test_df_with_minimal_results(minimal_res_dict):
+    with open("estimagic/tests/visualization/minimal_expected_df.json", "r") as f:
+        expected = pd.DataFrame(json.load(f))
+        expected["conf_int_upper"] = pd.np.nan
+        expected["conf_int_lower"] = pd.np.nan
+    res = _df_with_all_results(minimal_res_dict)
+    expected.set_index(["model", "full_name"], inplace=True)
+    res.set_index(["model", "full_name"], inplace=True)
+    assert_frame_equal(res, expected)
+
+
+def test_df_with_results_with_model_classes(res_dict_with_model_class):
+    with open(
+        "estimagic/tests/visualization/with_model_class_expected_df.json", "r"
+    ) as f:
+        expected = pd.DataFrame(json.load(f))
+        expected["conf_int_upper"] = pd.np.nan
```

Comment:
np.nan

Reply:
Changed in 0846b2d5aa68a1880f930eade981bce6930216aa.